device/cuda/src/sanity/contiguous_on.cuh

```diff
@@ -24,117 +25,127 @@
 
 // System include
 #include <concepts>
+#include <utility>
 
 namespace traccc::cuda {
+
 namespace kernels {
-template <std::semiregular P, typename T, typename S>
-requires std::regular_invocable<P, T> __global__ void compress_adjacent(
-    P projection, vecmem::data::vector_view<T> _in, S* out,
-    uint32_t* out_size) {
+
+/// Kernel used in implementing @c traccc::cuda::is_contiguous_on
+template <typename CONTAINER, std::semiregular P, typename VIEW,
+          std::equality_comparable S>
+requires std::regular_invocable<P, decltype(std::declval<CONTAINER>().at(0))>
+    __global__ void is_contiguous_on_compress_adjacent(
+        P projection, VIEW _in, vecmem::data::vector_view<S> out_view) {
+
     int tid = threadIdx.x + blockIdx.x * blockDim.x;
 
-    vecmem::device_vector<T> in(_in);
+    const CONTAINER in(_in);
+    vecmem::device_vector<S> out(out_view);
 
     if (tid > 0 && tid < in.size()) {
-        std::invoke_result_t<P, T> v1 = projection(in.at(tid - 1));
-        std::invoke_result_t<P, T> v2 = projection(in.at(tid));
+        S v1 = projection(in.at(tid - 1));
+        S v2 = projection(in.at(tid));
 
         if (v1 != v2) {
-            out[atomicAdd(out_size, 1u)] = v2;
+            out.push_back(v2);
         }
     } else if (tid == 0) {
-        out[atomicAdd(out_size, 1u)] = projection(in.at(tid));
+        out.push_back(projection(in.at(tid)));
```

Comment:
@stephenswat take note that I changed this code a little. Since you were literally implementing a resizable 1D vector by hand previously, I decided to rather just use the vecmem types here that were designed for exactly this use case. :thinking: